### DISCUSSION TITLE
Removing listener(s)

### DISCUSSION THREAD

**Discussion by Gaasduif** [leolabs/ableton-js#68] [Q&A]
I want to have a listener on a certain property of the selected clip. 
But if I add a listener to the property, it will keep firing events even when the clip not selected.
So I need to remove the listener when the clip is deselected and add a new listener for the currently selected clip.

What is the best way to do this?

I've already tried this removePropListener function for the 'playing_position'-property. 
When listening, I get the time value (like expected). After removing the listener, I still get the events, but the value is 0.
When selected the clip again, I get values again.
So, it seems to do something, but not like expected.

**Comment by leolabs** [comment]
Hey @CrossmodGlennVS,

The value returned by the `addListener` function is another function you can use to remove that listener. Basically:

```ts
// Subscribe to a value:
const unsubscribe = await ableton.song.addListener("tempo", (t) => console.log("Tempo:", t));

// And to unsubscribe:
await unsubscribe();
```

I hope this helps :)

**Reply by Gaasduif** [reply]
Thank you. It works.
